unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ExploreHUD/Scripts/BaseCellView.cs

```diff
@@ -1,30 +1,34 @@
 using System;
+using DCL;
 using DCL.Helpers;
 using UnityEngine;
 using UnityEngine.Networking;
 using UnityEngine.UI;
 
 internal class BaseCellView : MonoBehaviour
 {
-    [SerializeField] Image thumbnailImage;
+    [SerializeField] RawImageFillParent thumbnailImage;
     [SerializeField] GameObject loadingSpinner;
     [SerializeField] Sprite errorThumbnail;
 
-    public event Action<Sprite> OnThumbnailFetched;
+    public event Action<Texture2D> OnThumbnailFetched;
 
     UnityWebRequest thumbnailRequest = null;
     Texture2D thumbnailTexture;
-    Sprite thumbnail;
 
-    public void FetchThumbnail(string url)
+    public void FetchThumbnail(string url, Action onFetchFail)
```

Comment:
A bit confused on why we are passing `onFetchFail` as an argument, but we have `OnThumbnailFetched` as a member event.

Reply:
renamed `OnThumbnailFetched` to `OnThumbnailSet` cause the event is triggered when the thumbnail is set and not necessary means that the thumbnail was fetched successfully